drivers/sensor/bh1750/bh1750.c

```diff
@@ -0,0 +1,230 @@
+
+	return i2c_write_dt(&cfg->bus, &opcode, 1);
+}
+
+static int bh1750_mtreg_write(const struct device *dev, uint8_t mtreg)
+{
+	int rc;
+	uint8_t high_byte = mtreg & BH1750_MTREG_HIGH_BYTE_MASK;
+	uint8_t low_byte = mtreg & BH1750_MTREG_LOW_BYTE_MASK;
+
+	rc = bh1750_opcode_write(dev, BH1750_MTREG_HIGH_BYTE | (high_byte >> 5));
+	if (rc < 0) {
+		LOG_ERR("%s, Failed to write high byte of mtreg!",
+			dev->name);
+		return rc;
+	}
+
+	rc = bh1750_opcode_write(dev, BH1750_MTREG_LOW_BYTE | low_byte);
+	if (rc < 0) {
+		LOG_ERR("%s, Failed to write low byte of mtreg!",
+			dev->name);
+		return rc;
+	}
+
+	return 0;
+}
+
+static uint8_t bh1750_get_mode_from_dts_device(const struct device *dev)
+{
+	const struct bh1750_dev_config *cfg = dev->config;
+
+	if (cfg->resolution == BH1750_HIGH_RES_2_DTS_ENUM) {
+		return BH1750_ONE_TIME_HIGH_RES_MODE_2;
+	} else if (cfg->resolution == BH1750_HIGH_RES_DTS_ENUM) {
+		return BH1750_ONE_TIME_HIGH_RES_MODE;
+	} else {
+		return BH1750_ONE_TIME_LOW_RES_MODE;
+	}
+}
+
+static int bh1750_get_wait_time_from_dts_device(const struct device *dev)
+{
+	const struct bh1750_dev_config *cfg = dev->config;
+
+	if (cfg->resolution == BH1750_HIGH_RES_2_DTS_ENUM) {
+		return BH1750_HIGH_RES_MODE_MAX_WAIT;
+	} else if (cfg->resolution == BH1750_HIGH_RES_DTS_ENUM) {
+		return BH1750_HIGH_RES_MODE_MAX_WAIT;
+	} else {
+		return BH1750_LOW_RES_MODE_MAX_WAIT;
+	}
+}
+
+static int bh1750_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	const struct bh1750_dev_config *cfg = dev->config;
+	struct bh1750_data *drv_data = dev->data;
+	const int max_wait = bh1750_get_wait_time_from_dts_device(dev);
+	const uint8_t mode = bh1750_get_mode_from_dts_device(dev);
+	uint16_t value;
+	int rc;
+	int wait_time;
+
+	if (chan != SENSOR_CHAN_ALL &&
+	    chan != SENSOR_CHAN_LIGHT) {
+		return -ENOTSUP;
+	}
+
+	/* clear sensor values */
+	drv_data->sample = 0U;
+
+	/* Start the measurement */
+	rc = bh1750_opcode_write(dev, mode);
+	if (rc < 0) {
+		LOG_ERR("%s, Failed to start measurement!",
+			dev->name);
+		return rc;
+	}
+
+	/* Calculate measurement time */
+	wait_time = (max_wait * (cfg->mtreg * 10000 / BH1750_DEFAULT_MTREG)) / 10000;
+
+	/* Wait for the measurement to be stored in the sensor memory */
+	k_msleep(wait_time);
+
+	/* Fetch result */
+	rc = bh1750_opcode_read(dev, mode, &value);
+	if (rc < 0) {
+		LOG_ERR("%s: Failed to read measurement result!",
+			dev->name);
+		return rc;
+	}
+
+	/* Store measurements */
+	drv_data->sample = value;
+
+	return 0;
+}
+
+static int bh1750_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	const struct bh1750_dev_config *cfg = dev->config;
+	struct bh1750_data *drv_data = dev->data;
+	uint32_t tmp;
+
+	if (chan != SENSOR_CHAN_ALL &&
+	    chan != SENSOR_CHAN_LIGHT) {
+		return -ENOTSUP;
+	}
+
+	/* See datasheet for more information how to convert raw sample to lx */
```

Comment:
Please provide a link and section number

Reply:
Should I paste the link into code? If so, is it ok to link to lets say component supplier like mouser? Unfortunately from what I see the datasheet is no longer available at the manufacturer website.

https://www.mouser.com/datasheet/2/348/Rohm_11162017_ROHMS34826-1-1279292.pdf
Page 11

Reply:
Yes, please. The original source would be better, but if that's not available then Mouser is fine.